src/main.cpp

```diff
@@ -1502,11 +1508,30 @@ MempoolReturnValue AcceptTxToMemoryPool(CTxMemPool& pool, CValidationState &stat
             return MempoolReturnValue::INVALID;
         }
 
+        if (tx.GetVcswCcIn().empty() && tx.GetVBwtRequestOut().empty())
+        {
+            //No sc proofs to be validated, job done. Store and exit
+            StoreTxToMempool(tx, pool, view);
+            return MempoolReturnValue::VALID;
+        }
+
         if (validateScProofs == ValidateSidechainProof::OFF)
             return MempoolReturnValue::PARTIALLY_VALIDATED;
 
         CScProofVerifier scVerifier{CScProofVerifier::Verification::Strict};
-        if (!view.CheckScTxProof(tx, scVerifier))
+
+        scVerifier.LoadDataForMbtrVerification(view, tx);
+        std::map<uint256, bool> resMbtr = scVerifier.batchVerifyMbtrs();
+        if (resMbtr.count(hash) != 0 && !resMbtr.at(hash))
+        {
+            state.Invalid(error("%s():%d - ERROR: sc-related tx [%s] proofs do not verify\n",
+                          __func__, __LINE__, hash.ToString()), REJECT_INVALID, "bad-sc-tx-proof");
+            return MempoolReturnValue::INVALID;
+        }
+
+        scVerifier.LoadDataForCswVerification(view, tx);
+        std::map<uint256, bool> resCsw = scVerifier.batchVerifyCsws();
+        if (resCsw.count(hash) != 0 && !resCsw.at(hash))
```

Comment:
maybe `if (resCsw.count(hash) == 0 || !resCsw.at(hash))`
to cover the potential issue caused by scVerifer returning empty `resCsw` or `resCsw` with no `hash`.

Same for mbtrs.

Reply:
Actually `resCsw.count(hash) == 0` covers the case of a tx with bwts but not csw. If I replace `&&` with `||` I would flag all such txes as invalid. We may implement your suggestion for bwts once we remove proof for csws